migrations/frontend/1528395863_create_temporary_settings.up.sql

```diff
@@ -0,0 +1,19 @@
+BEGIN;
+
+CREATE TABLE IF NOT EXISTS temporary_settings (
+    id serial NOT NULL PRIMARY KEY,
+    user_id integer NOT NULL UNIQUE,
+    contents text,
```

Comment:
You can make this a `json` or `jsonb` column type which will make it easy to query from that column in the database layer if you ever need to :) 

```sql
-- Returns the number of users which have completed the search tour. Very nice for metrics.
SELECT COUNT(id) FROM temporary_settings WHERE contents->'hasSearchTourCompleted' IS TRUE;
```

```suggestion
    contents JSONB,
```

Reply:
Settings are JSONC, I don't think Postgres supports that unfortunately 😞

Reply:
Don't think temporary settings need to be JSONC. We get a lot by making this column jsonb

Reply:
>We get a lot by making this column jsonb

True and agreed.

I don't know how temporary settings are supposed to work inside the product, @limitedmage is there a doc I can read?

If we store as JSONB, we basically strip out all comments, and these comments (since users can use them already) are not all useless.

Reply:
The RFC is [here](https://docs.google.com/document/d/1PCG1BhagmALYBEUe9PF7I_0W573Fodw4lOEbhYZCyuI/edit#heading=h.trqab8y0kufp) -- based on my reading of it, I don't think temporary settings are ever meant to be user-editable, so comments would not be necessary

Reply:
> The RFC is [here](https://docs.google.com/document/d/1PCG1BhagmALYBEUe9PF7I_0W573Fodw4lOEbhYZCyuI/edit#heading=h.trqab8y0kufp) -- based on my reading of it, I don't think temporary settings are ever meant to be user-editable, so comments would not be necessary

@camdencheek  - thanks, but how do we restore settings?

Reply:
Just to make sure we're on the same page, temporary settings and user settings are disjoint concepts. My (somewhat limited) understanding is that the goal of temporary settings is essentially to replace `LocalStorage` in a way that follows a logged-in user across browsers. So, for example, if I dismiss a popup on my work laptop, then pop over to my phone, that same popup doesn't show back up because it was marked as dismissed in the temporary settings table (rather than just locally in my browser). 

These types of settings aren't meant to be user-editable, or even user-visible, and I don't think we need to support restoring them either.

Reply:
@camdencheek Ah! I see, OK. Let's use JSONB!